### PR TITLE
Make 24 hour or AM/PM optional

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR makes a 24 hour or AM/PM type clock optional

### CODE REVIEW

**Comment by stefanv** [comment]
Could you explain the goal of this PR is some more detail? What's the motivation for not always using 24h?

**Comment by mcoughlin** [comment]
@stefanv Right now DateTimePicker defaults to AM/PM, and there is a request for 24 Hours. maybe because I am American I find AM/PM a bit more natural.

File: static/js/components/UIPreferences.jsx [hunk 1]
```diff
@@ -13,6 +13,7 @@ const UIPreferences = () => {
   const currentTheme = preferences?.theme;
   const invertThumbnails = preferences?.invertThumbnails || false;
   const compactComments = preferences?.compactComments || false;
+  const twelveOrTwentyFour = preferences?.twelveOrTwentyFour || false;
```
Comment (guynir42) [inline static/js/components/UIPreferences.jsx:16]: What are the values going to be? `true` or `false`? Which corresponds to 24h and which to am/pm?

**Review by guynir42 — APPROVED** [review]
I would make it a bit more obvious in the code (with a comment or a better variable name) what values represent 24h vs. am/pm. Other than that it seems fine.

File: static/js/components/UIPreferences.jsx (hunk 1 above)
Comment (mcoughlin) [reply static/js/components/UIPreferences.jsx]: yes making this more obvious in the variable name now :)

**Review by guynir42 — APPROVED** [review]

**Pull request merged**